sklearn/linear_model/ransac.py

```diff
@@ -195,6 +195,10 @@ def fit(self, X, y):
             `is_data_valid` and `is_model_valid` return False for all
             `max_trials` randomly chosen sub-samples.
 
+        TypeError
```

Comment:
Also, can you remove this block and document it under `sample_weight`?, i.e

    sample_weight: array-like, 
               blah, blah.. raises Error if balh blah